tests/test_utils.py

```diff
@@ -34,3 +41,19 @@ def test_mmh_to_mmtimestep():
     expected_mmtimestep_series = [0.3, 0.6, 0.9, 1.2, 1.5, 2.1, 2.7, 3.3, 3.3, 2.1, 1.2, 0.6, 0.0]
     mmtimestep_series = [round(mmh_to_mmtimestep(v, timestep, units), 2) for v in mmh_series]
     assert mmtimestep_series == expected_mmtimestep_series
+
+
+def test_extract_error_message():
+    sim_error = SimpleApiException(SIM_EXCEPTION_BODY)
+    detailed_error = SimpleApiException(DETAILED_EXCEPTION_BODY)
+    rel_error = SimpleApiException(RELATED_OBJECTS_EXCEPTION_BODY)
+    decode_error = SimpleApiException("invalid JSON body")
+    unknown_error_body = SimpleApiException(["nobody expects the spanish inquisition"])
+    assert extract_error_message(sim_error) == "Error: {'duration': ['minimum value for duration is 60']}"
+    assert extract_error_message(detailed_error) == "Error: {'duration': ['minimum value for duration is 60']}"
+    assert (
+        extract_error_message(rel_error)
+        == "Error: File processing error: Invalid file, see related event for further details (<related_object>)"
+    )
+    assert extract_error_message(decode_error) == "Error: invalid JSON body"
+    assert extract_error_message(unknown_error_body) == "Error: ['nobody expects the spanish inquisition']"
```

Comment:
It is best to split it into five separate tests. Clearer if one of them fails.

Bonus points for the last error message, btw :-)

Reply:
Tests separated.